lib/v2/heu/ugs/news.js

```diff
@@ -0,0 +1,122 @@
+        wytk: '/2897',
+        cjgl: '/2901',
+    },
+    sjjxyjlc: {
+        all: '/3206',
+        syjx: '/2847',
+        sysjs: '/sysjs',
+        xwsx: '/2909',
+        xwlw: '/2910',
+        kcsj: '/2911',
+        cxcy: '/2913',
+        xjjl: '/xjjl',
+    },
+    jypgc: {
+        all: '/3207',
+        jxyjyjxcg: '/2916',
+        zljk: '/2917',
+    },
+    zyjsc: {
+        all: '/3208',
+        zyyjcjs: '/2914',
+        cgsyb: '/2925',
+        jxmsyyxzjjs: '/2918',
+        ktjs: '/2919',
+        syjx: '/2920',
+    },
+    gjdxswhszjd: {
+        all: '/3209',
+    },
+    jsjxfzzx: {
+        all: '/3210',
+        jspx: '/2915',
+    },
+    zhbgs: {
+        all: '/3211',
+        lxkc: '/lxkc',
+    },
+};
+
+module.exports = async (ctx) => {
+    const author = ctx.params.author || 'gztz';
+    const category = ctx.params.category || 'all';
+    const link = baseUrl + authorMap[author][category] + '/list.htm';
+    const response = await got({
+        method: 'get',
+        url: link,
+        headers: {
+            Referer: baseUrl,
+        },
+    });
+    const $ = cheerio.load(response.data);
+
+    const urlList = $('.wp_article_list_table .border9')
+        .slice(0, 10)
+        .map((i, e) => $('a', e).attr('href'))
+        .get();
+
+    const titleList = $('.wp_article_list_table .border9')
+        .slice(0, 10)
+        .map((i, e) => $('a', e).text())
+        .get();
+
+    const dateList = $('.wp_article_list_table .border9')
+        .slice(0, 10)
+        .map((i, e) => $('.date', e).text())
+        .get();
+
+    const out = await Promise.all(
+        urlList.map(async (itemUrl, index) => {
+            itemUrl = url.resolve(baseUrl, itemUrl);
+            if (itemUrl.indexOf('.htm') !== -1) {
+                const cache = await ctx.cache.get(itemUrl);
+                if (cache) {
+                    return Promise.resolve(JSON.parse(cache));
+                }
+                const response = await got.get(itemUrl);
+                const $ = cheerio.load(response.data);
+                const single = {
+                    title: titleList[index],
+                    link: itemUrl,
+                    description: $('.wp_articlecontent')
+                        .html()
+                        .replace(/src="\//g, `src="${url.resolve(baseUrl, '.')}`)
+                        .replace(/href="\//g, `href="${url.resolve(baseUrl, '.')}`)
+                        .trim(),
+                    pubDate: dateList[index],
+                };
+                ctx.cache.set(itemUrl, JSON.stringify(single));
+                return Promise.resolve(single);
+            } else {
+                const single = {
+                    title: titleList[index],
+                    link: itemUrl,
+                    description: '此链接为文件，请点击下载',
+                    pubDate: dateList[index],
+                };
+                return Promise.resolve(single);
+            }
+        })
+    );
```

Comment:
- `url.resolve()` is deprecated
- Please use `ctx.cache.tryGet()`
```suggestion
    const out = await Promise.all(
        list.map((item) =>
            ctx.cache.tryGet(item.link, async () => {
                if (item.link.indexOf('.htm') !== -1) {
                    const response = await got(item.link);
                    const $ = cheerio.load(response.data);

                    item.description = $('.wp_articlecontent')
                        .html()
                        .replace(/src="\//g, `src="${new URL('.', baseUrl).href}`)
                        .replace(/href="\//g, `href="${new URL('.', baseUrl).href}`)
                        .trim();
                } else {
                    item.description = '此链接为文件，请点击下载';
                }
                return item;
            })
        )
    );
```